json_transport/src/json_transport/json_transport.py

```diff
@@ -0,0 +1,50 @@
+# Software License Agreement (BSD)
+#
+# \file      json_transport.py
+# \authors   Paul Bovbel <pbovbel@locusrobotics.com>
+# \copyright Copyright (c) (2018,), Locus Robotics, All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are permitted
+# provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+# and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+# conditions and the following disclaimer in the documentation and/or other materials provided with
+# the distribution.
+# 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+# endorse or promote products derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+# FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+# OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import msgpack
+import rospy
+from json_msgs import msg as json_msg
+
+
+class PackedJson(rospy.msg.AnyMsg):
+
+    _md5sum = json_msg.Json._md5sum
+    _type = json_msg.Json._type
+
+    def __init__(self, *args, **kwargs):
+        if len(args) == 1:
+            self.data = args[0]
+        elif args:
+            self.data = args
+        elif kwargs:
+            self.data = kwargs
+
+    def set_data(self, data):
+        self._buff = msgpack.packb(data)
+
+    def get_data(self):
+        return msgpack.unpackb(self._buff)
+
+    data = property(get_data, set_data)
```

Comment:
Don't like @property decorators?

Reply:
I think this style is a little more explicit, I've always found the decorators to be a little wonky looking.